src/Integration/NewConnectedMode/ConfigurationProvider.cs

```diff
@@ -68,18 +101,46 @@ public bool WriteConfiguration(BindingConfiguration configuration)
                 throw new ArgumentNullException(nameof(configuration));
             }
 
+            var onSuccessfulFileWrite = GetOnSuccessfulFileWriteOperation(configuration);
+            
+            if (onSuccessfulFileWrite == null)
+            {
+                return false;
+            }
+
+            GetBoundProject(out _, out var bindingPath);
```

Comment:
See my comment on "WriteConfig_NewConnectedModeConfig_SaveNewConfig". I don't this logic is correct; if I'm reading it correctly, GetBound is not just calculating the path, it's also checking for the existence of the file.
When saving, we shouldn't care whether the file already exists or not (otherwise what will happen the first time we try to save a new config file in connected mode?)

Reply:
https://github.com/SonarSource/sonarlint-visualstudio/pull/1252/files#diff-7f61f05d37ffa1d26cbeb2018908ee31
wdyt?